app/code/Magento/QuoteGraphQl/Model/Resolver/AddSimpleProductsToCart.php

```diff
@@ -81,6 +81,7 @@ public function resolve(Field $field, $context, ResolveInfo $info, array $value
 
         $this->addProductsToCart->execute($cart, $cartItems);
         $cartData = $this->extractDataFromCart->execute($cart);
+        $cartData["cart_id"] = $cartHash;
```

Comment:
Hello @dominicfernando,
It's should be part of `ExtractDataFromCart` 
If `ExtractDataFromCart` will return `cart_id` then we will cover all of the cases (like adding different types of products)

We already have this fix in PR
https://github.com/magento/graphql-ce/pull/320/files#diff-b5e7e8d0c42394996dc70fa755840759R61